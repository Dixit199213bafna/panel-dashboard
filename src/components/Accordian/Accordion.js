import React, { useState } from 'react';
import classes from './Accordian.module.css';
import Feature from '../../components/Feature/Feature'
const Accordion = (props) => {
  const [isActive, setIsActive] = useState(false);

  const onHeaderToggleHandler = (e) => {
      setIsActive(e.value);
      props.onHeaderToggleClick(e)
  }
  return (
    <div className="accordion-item">
      <div className={classes.AccordianTitle} >
        <Feature {...props} accordionHeader="true" onToggleClick={(e) => onHeaderToggleHandler(e)}/>
        <div className='toggle' onClick={() => setIsActive(!isActive)}>{isActive ? <span className={classes.DownCaret}/> : <span className={classes.UpCaret}/>}</div>
      </div>
      {isActive && <React.Fragment><div className={classes.AccordionContent}>
                        {props.childFeatureList.map((feat, index) => <Feature  key={feat.featureId} accordionHeader="true" panel={props.panel} {...feat} onToggleClick={(e) => props.onFeatureChildToggleClick({
                            ...e,
                            childPanelId: props.featureId
                        })} onValueChange={(e) => props.onChildValueChange({
                            ...e,
                            childPanelId: props.featureId
                        })}/>)}  
                    </div></React.Fragment>}
    </div>
  );
};

export default Accordion;