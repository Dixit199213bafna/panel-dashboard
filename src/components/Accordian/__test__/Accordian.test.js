import React from 'react';
import renderer from 'react-test-renderer';
import panelList from '../../../mock/panel-list.mock';
import Accordion from '../Accordion';
import TestRenderer from 'react-test-renderer';
const {act} = TestRenderer;

describe('Accordion', () => {
  let tree;
  const props = panelList['Settings'].featureList[1];
  const component = renderer.create(
    <Accordion {...props}/>,
  );

  const instance = component.getInstance();

  it('Accordion Initialized', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Accordion Toggle', async () => {
    act(() => {
        component.root.findByProps({ className: "toggle" }).props.onClick();
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


});