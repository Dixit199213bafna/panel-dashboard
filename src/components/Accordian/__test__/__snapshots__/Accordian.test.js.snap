// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Accordion Initialized 1`] = `
<div
  className="accordion-item"
>
  <div
    className="AccordianTitle"
  >
    <div
      className="FeatureAccordionHeader"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          USERS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={22}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="toggle"
      onClick={[Function]}
    >
      <span
        className="UpCaret"
      />
    </div>
  </div>
</div>
`;

exports[`Accordion Accordion Toggle 1`] = `
<div
  className="accordion-item"
>
  <div
    className="AccordianTitle"
  >
    <div
      className="FeatureAccordionHeader"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          USERS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={22}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="toggle"
      onClick={[Function]}
    >
      <span
        className="DownCaret"
      />
    </div>
  </div>
  <div
    className="AccordionContent"
  >
    <div
      className="FeatureAccordionHeader"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          USERS - Add
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={221}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="FeatureAccordionHeader"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          USERS - Edit
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={222}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="FeatureAccordionHeader"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          USERS - Delete
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={223}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="FeatureAccordionHeader"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          Max USERS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <select
          className="Select"
          onChange={[Function]}
          value={5}
        >
          <option
            value={5}
          >
            5
          </option>
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={50}
          >
            50
          </option>
        </select>
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={224}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
